### PR TITLE
www-client/netsurf & friends: bump to latest version

### PR DESCRIPTION
**Pull request by trilean**
Bump NetSurf to version 3.9 and its dependencies to the latest version: http://source.netsurf-browser.org/

Since the ebuild is currently unmaintained I'd like to step up as a maintainer.

### CODE REVIEW

**Comment by trilean** [comment]
Ooops forgot the signoff

**Comment by juippis** [comment]
```
LD_LIBRARY_PATH=build/Linux-test/ build/Linux-test/bloom
Running suite(s): Bloom filter
100%: Checks: 230, Failures: 0, Errors: 0
50%: Checks: 4, Failures: 2, Errors: 0
test/bloom.c:56:S:Match:bloom_match_test:0: Assertion 'dictf != ((void *)0)' failed
test/bloom.c:56:S:False positive rate:bloom_falsepositive_test:0: Assertion 'dictf != ((void *)0)' failed
make: *** [test/Makefile:201: bloom_test] Error 1
 * ERROR: www-client/netsurf-3.9::testworld failed (test phase):
 *   emake failed
```

Merges fine with tests disabled, and runs fine too. You'll either need to restrict tests, or fix them. 

Also you can add yourself as proxy-maintainer by modifying the metadata.xml file, in this or some other PR as you wish. 
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml

File: media-libs/libnspsl/libnspsl-0.1.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+HOMEPAGE="http://www.netsurf-browser.org/projects/"
+SRC_URI="https://download.netsurf-browser.org/libs/releases/${P}-src.tar.gz"
+
+LICENSE="MIT"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~m68k-mint"
+IUSE=""
+
+DEPEND="
+	dev-util/netsurf-buildsystem
```
Comment (juippis) [inline media-libs/libnspsl/libnspsl-0.1.5.ebuild]: Looking at the content of this package, I think this should go in DEPEND and BDEPEND.

File: net-libs/libhubbub/libhubbub-0.3.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+RDEPEND="
+	>=dev-libs/libparserutils-0.2.1-r1
+	!net-libs/hubbub"
+DEPEND="${RDEPEND}
+	dev-util/netsurf-buildsystem
+	virtual/pkgconfig
+	test? (
+		dev-lang/perl
+		<dev-libs/json-c-0.13
```
Comment (juippis) [inline net-libs/libhubbub/libhubbub-0.3.6.ebuild]: Are you sure about this requirement? Did you check if it was updated? 
(Caused me some downgrades)

**Comment by juippis** [comment]
Oh and by the way, you can automate the signing off by adding `DCO_SIGNED_OFF_BY` line to make.conf when committing with repoman. Check `man 5 make.conf` for more info.

**Comment by kripton** [comment]
Hi, I've tested your PR and the resulting netsurf-3.9 on powerpc-unknown-linux-gnu (ppc arch). Builds fine and runs much better than the 3.8* in the tree. Thanks for your work!

File: net-libs/libhubbub/libhubbub-0.3.6.ebuild (hunk 2 above)
Comment (trilean) [reply net-libs/libhubbub/libhubbub-0.3.6.ebuild]: Yes but the fix was easy. It's included as a patch now. I'll send it upstream.

File: media-libs/libnspsl/libnspsl-0.1.5.ebuild (hunk 1 above)
Comment (trilean) [reply media-libs/libnspsl/libnspsl-0.1.5.ebuild]: Sure? Why both?

File: dev-libs/libcss/libcss-0.9.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~arm ~ppc ~ppc64 ~m68k-mint"
+IUSE="test"
+
+RDEPEND="
+	>=dev-libs/libparserutils-0.2.1-r1
+	>=dev-libs/libwapcaplet-0.4.0"
+DEPEND="${RDEPEND}
+	>=dev-util/netsurf-buildsystem-1.7-r1
+	virtual/pkgconfig
```
[the remark this answers is not in the document]
Comment (trilean) [reply dev-libs/libcss/libcss-0.9.0.ebuild]: Done for all ebuilds.

**Comment by trilean** [comment]
> ```
> [...]
>  * ERROR: www-client/netsurf-3.9::testworld failed (test phase):
>  *   emake failed
> ```
> 
> Merges fine with tests disabled, and runs fine too. You'll either need to restrict tests, or fix them.

I've restricted the tests for now. Upstream hasn't changed the bloom tests in years and besides I can't even get the tests to build. Probably a glibc issue, I'll look into it.

**Comment by trilean** [comment]
> ## Pull request CI report
> [...]
> New issues caused by PR:
> https://qa-reports.gentoo.org/output/gentoo-ci/30c1d80/output.html#www-client/netsurf

That's a false alarm?

File: net-libs/libhubbub/libhubbub-0.3.6.ebuild (hunk 2 above)
Comment (juippis) [reply net-libs/libhubbub/libhubbub-0.3.6.ebuild]: If it gets executed during the build, its a BDEPEND. If it has libraries/headers that are needed during build, its a DEPEND. Granted I dont know what this package does exactly, but looked like it possibly had both.

**Comment by juippis** [comment]
> 
> That's a false alarm?

You seem to be dropping bunch of those locally defined flags in your last commit, should be an easy fix ;)

**Comment by juippis** [comment]
All good, until:
```
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: build/Linux-gtk3/build_Linux-gtk3_duktape_binding.o: in function `dukky_create_prototype':
/tmp/portage/www-client/netsurf-3.9/work/netsurf-3.9/build/Linux-gtk3/duktape/binding.c:113: undefined reference to `LOG'
collect2: error: ld returned 1 exit status
make: *** [Makefile:712: nsgtk3] Error 1
```
I can't emerge netsurf with `duktape` USE on. It emerges fine with `USE="-duktape"`. Any ideas?

**Comment by trilean** [comment]
> ```
> /usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: build/Linux-gtk3/build_Linux-gtk3_duktape_binding.o: in function `dukky_create_prototype':
> /tmp/portage/www-client/netsurf-3.9/work/netsurf-3.9/build/Linux-gtk3/duktape/binding.c:113: undefined reference to `LOG'
> collect2: error: ld returned 1 exit status
> make: *** [Makefile:712: nsgtk3] Error 1
> ```
That line should likely be something like `NSLOG(dukky, WARNING, ...)` and not `LOG(...)`. I need to check how they do it in HEAD and if there's a patch for that.

**Comment by trilean** [comment]
@juippis your error is fixed. It turns out bumping `dev-libs/nsgenbind` was sufficient :wink:

http://source.netsurf-browser.org/nsgenbind.git/commit/?id=9656f1ff87136263d44a54a4e3081ac92ad3b699

**Comment by juippis** [comment]
Thanks, I did test it, but didn't dig deeper into what you had done. But yes, it works great.
Pushed an update, 
fd23ea8411ac101d22aabeba2a1c2c0d157ea786

**Pull request closed without merging**